rdkcentral/Dobby
# RDK-30110: Build in release mode without errors

### Description
Allow Dobby to build in Release mode without any errors. Set the log levels appropriately. 

Note release builds do not include DobbyTool or DobbyBundleGenerator.

Also renames the Xi6 platform to `Generic` as Dobby Spec support is not required on Xi6 platforms so no extra settings are needed specific to Xi6. 

Legacy Components default to OFF

Adds a new CMake option for use by Bitbake to append the SRC_REV to the version number of Dobby for easy checking of build version on the STB

### Test Procedure
Build Dobby in release mode both with and without LTO enabled. Should build without errors and no regressions should be visible in Sky and generic RDK builds

### Type of Change

### Requires Bitbake Recipe changes?
